README.md

```diff
@@ -21,44 +21,150 @@
+```shell
+./.nix/nix-cmake.sh \
+  -DCMAKE_BUILD_TYPE=Debug \
+  -G Ninja \
+  -S . -B cmake-build-debug
+
+./.nix/nix-cmake.sh --build cmake-build-debug
+./.nix/ctest --test-dir cmake-build-debug -j
+```
+
+Generate IDE tool shims when you need CLion to re-enter the shell automatically:
+
+```shell
+nix run .#clion-setup
+```
+
+#### Use the development container
+Build or reuse the local development image (installs the current user inside the container to avoid permission issues):
+
+```shell
+./scripts/install-local-docker-environment.sh
+```
+
+Configure, build, and test by mounting the repository into the container:
+
+```shell
+docker run \
+  --workdir $(pwd) \
+  -v $(pwd):$(pwd) \
+  nebulastream/nes-development:local \
+  cmake -B build-docker
+
+docker run \
+  --workdir $(pwd) \
+  -v $(pwd):$(pwd) \
+  nebulastream/nes-development:local \
+  cmake --build build-docker -j
+
+docker run \
+  --workdir $(pwd) \
+  -v $(pwd):$(pwd) \
+  nebulastream/nes-development:local \
+  ctest --test-dir build-docker -j
+```
+
+### Execute Queries & Tests
+
+#### Run systest for a query
+Build the `systest` executable in your chosen build directory and execute a specific query. The example below targets the first query in the arithmetic function suite and stores artifacts in `cmake-build-debug/systest-run`:
+
+```shell
+BUILD_DIR=cmake-build-debug
+cmake --build $BUILD_DIR -j --target systest
+$BUILD_DIR/systest/systest -t nes-systests/function/arithmetical/FunctionAdd.test:1
+```
+
+Append worker configuration overrides after `--` (for example `-- --worker.default_query_execution.execution_mode=INTERPRETER`). See the [systest guide](docs/development/systests.md) for authoring custom `.test` files and the complete CLI reference.
+
+#### Start nebuli with a worker
+Build the coordinator and worker binaries, start the worker in one terminal, and submit a short-lived query from another terminal using `nebuli`:
+
+```shell
+BUILD_DIR=cmake-build-debug
+cmake --build $BUILD_DIR -j --target nes-single-node-worker nes-nebuli
+
+# Terminal 1: start the worker (listens on localhost:8080 by default)
+$BUILD_DIR/nes-single-node-worker/nes-single-node-worker
+
+# Terminal 2: prepare and register a demo query
+cat > demo.sql <<EOF
+CREATE LOGICAL SOURCE demo(value UINT64);
+CREATE PHYSICAL SOURCE FOR demo TYPE File;
+ATTACH INLINE
+1
+2
+3
+
+CREATE SINK result(value UINT64) TYPE File SET('$(pwd)/demo-output.csv' as `SINK`.FILE_PATH, 'CSV' as `SINK`.INPUT_FORMAT);
+SELECT value FROM demo INTO result;
+EOF
+
+$BUILD_DIR/nes-nebuli/nes-nebuli register \
+  -s localhost:8080 \
+  -i demo.sql \
+  -x \
+  -w
+```
+
+The `-x` flag starts the query immediately after registration and `-w` waits for completion. The generated CSV appears at `demo-output.csv`. Inspect or retire the query with additional `nebuli` subcommands such as `show`, `stop`, or `unregister` as needed.
```

Comment:
This definitely does not work. Have you tested these? There is `ATTACH INLINE? in the SQL frontend. There is no -x flag for the SQL frontend either.